packages/react/src/views/ChatBody/ChatBody.js

```diff
@@ -182,7 +192,24 @@ const ChatBody = ({
   const showNewMessagesPopup = () => {
     setPopupVisible(true);
   };
+  const announcementRef = useRef(null);
+
+  const toggleModal = () => {
+    setModalOpen(!isModalOpen);
+  };
+
+  const checkOverflow = () => {
```

Comment:
Why is this needed ?

Reply:
I have added that so that modal display only works if text overflows

Reply:
Instead of checking it like this, can we find out some other way ? 🤔

Reply:
I think we can set a threshold for the number of characters, but will that be consistent across different screen sizes?

Reply:
Yes you are correct, i was thinking of the same.. ok for now let's go with this...